src/main/kotlin/com/dropbox/mobile/hypershard/Hypershard.kt

```diff
@@ -125,20 +128,22 @@ class RealHyperShard(
                 val types = cu.types
                 for (type in types) {
                     val classOrInterfaceDeclaration =
-                            type as? ClassOrInterfaceDeclaration ?: continue
-
-                    val annotationUiTest =
-                            classOrInterfaceDeclaration.getAnnotationByName(annotationName)
-                    if (annotationUiTest.isPresent) {
+                        type as? ClassOrInterfaceDeclaration ?: continue
+                    val shouldProcessTestMethod =
+                        shouldProcessTestMethods(classOrInterfaceDeclaration, annotationValue)
+                    if (shouldProcessTestMethod) {
                         val methods = type.methods
                         for (method in methods) {
                             val annotationTest = method.getAnnotationByName("Test")
                             if (annotationTest.isPresent) {
                                 tests.add(
-                                        String.format("%s.%s.%s",
-                                                cu.packageDeclaration.get().name,
-                                                type.name,
-                                                method.name))
+                                    String.format(
```

Comment:
nitpick Kotlin has native text interpolation, yous hould be able to do "$cu.packageDeclaration.get().name.$type.name,$method.name"